### PR TITLE
[refactor] HM and GM

### LINKED ISSUE #212: [Refactor] Remove player states condition in GM
Breakdown InvokePlayerState function to several functions

### PR DESCRIPTION
**Pull request by naufalhilmi12345**
### Main changes

- Refactor GM
- Refactor connection between GM and HM
- Refactor ExorcismBar.cs

### Tasks List

- [close #212]

### Attachments
All test is passed:
image

### CODE REVIEW

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 1]
```diff
@@ -66,28 +56,27 @@ public void InvokeChangeWorld()
...
+        PlayerActionMapEvent?.Invoke("Hiding");
+    }
+
+    public void InvokeExorcismState()
+    {
+        HudMapEvent?.Invoke("Exorcism", true);
+        PlayerActionMapEvent?.Invoke("Exorcism");
+    }
+
+    public void InvokePlayerState()
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:76]: Rename function to: ResetPlayerState()

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 2]
```diff
@@ -66,28 +56,27 @@ public void InvokeChangeWorld()
         ChangeWorldEvent?.Invoke(_isInAstralWorld);
     }
 
-    public void InvokePlayerState(String state)
+    public void InvokeDialogueState()
+    {
+        HudMapEvent?.Invoke("Dialogue", true);
+        PlayerActionMapEvent?.Invoke("Dialogue");
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:62]: Rename event PlayerActionMapEvent to: PlayerStateEvent

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 3]
```diff
@@ -13,44 +13,34 @@ public class GameManager : MonoBehaviour, IGameManager
...
-        Default
-    }
     #endregion
 
     #region Event
     public static event Action<bool> ChangeWorldEvent;
     public static event Action<string> PlayerActionMapEvent;
-    public static event Action<bool> ShowDialogueHudEvent;
-    public static event Action<bool> StartHidingHudEvent;
+    public static event Action<string, bool> HudMapEvent;
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:21]: Rename event HudMapEvent to HudEvent

File: Assets/Astralization/Scripts/Managers/HudManager.cs [hunk 4]
```diff
@@ -21,25 +21,33 @@ public class HudManager : MonoBehaviour
...
-        GameManager.StartHidingHudEvent -= ShowHidingHud;
+        GameManager.HudMapEvent -= SetActiveHud;
         DialogueInputHandler.NextDialogueHudEvent -= NextDialogue;
         HideInputHandler.StopHidingHudEvent -= ShowHidingHud;
         Inventory.ItemLogoEvent -= UpdateLogo;
-        GameManager.ShowExorcismHudEvent -= ShowExorcism;
     }
     #endregion
 
+    public void SetActiveHud(string hudMap, bool state)
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/HudManager.cs:39]: Rename function: SetHudState(string hudKey, bool condition)

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 5]
```diff
@@ -66,28 +56,27 @@ public void InvokeChangeWorld()
         ChangeWorldEvent?.Invoke(_isInAstralWorld);
     }
 
-    public void InvokePlayerState(String state)
+    public void InvokeDialogueState()
+    {
+        HudMapEvent?.Invoke("Dialogue", true);
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:61]: strings that represents state name can be refactored. store this string in a static const field inside each state class. Variable name for this field is "name"

**Review by rafiftaris — CHANGES_REQUESTED** [review]

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 6]
```diff
@@ -66,28 +56,27 @@ public void InvokeChangeWorld()
...
+    }
+    public void InvokeHidingState()
+    {
+        HudMapEvent?.Invoke("Hiding", true);
+        PlayerActionMapEvent?.Invoke("Hiding");
+    }
+
+    public void InvokeExorcismState()
+    {
+        HudMapEvent?.Invoke("Exorcism", true);
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:72]: Make a single function: SendHudEvent(string, bool) to specifically call the invoke method. Use in all other methods

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 7]
```diff
@@ -66,28 +56,27 @@ public void InvokeChangeWorld()
...
+        PlayerActionMapEvent?.Invoke("Exorcism");
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:73]: Similarly, make a single function: SendPlayerStateEvent(string) to specifically call the invoke method. Use in all other methods

File: Assets/Astralization/Scripts/Managers/GameManager.cs [hunk 8]
```diff
@@ -66,28 +56,27 @@ public void InvokeChangeWorld()
         ChangeWorldEvent?.Invoke(_isInAstralWorld);
     }
 
-    public void InvokePlayerState(String state)
+    public void InvokeDialogueState()
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Managers/GameManager.cs:59]: Create regions for InvokeFunctions and SendEvents

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
Some changes

**Review by JahnsMichael — APPROVED** [review]
there is no issue from me. good job!

**Review by raihansyahdean — APPROVED** [review]
Great work

**Review by rafiftaris — APPROVED** [review]
mantapp no problem

**Pull request merged**